### PR TITLE
Adjust json output node order

### LINKED ISSUE #214: for output on coordinates, return output in the same order as the given nodal coordinates
for output on coordinates, return output in the same order as the given nodal coordinates

currently output is given in a sorted node id order

### PR DESCRIPTION
**Pull request by morettid**
- test order of mesh nodes matches the desired order of the required output nodes (coordinates)
- coarsen the 3D test for the output on a line of coordinates
- add adjustments for the json output so that the order of the key in the json file matches the nodes at the required coordinates (Kratos sorts them automatically).

Resolves #214

### CODE REVIEW

File: tests/test_stem.py [hunk 1]
```diff
@@ -420,6 +422,37 @@ def test_run_calculation_with_valid_stages(self, create_default_model: Model):
         # check if finalise is called
         stem.finalise.assert_called_once()
 
+    def test_validation_finalisation_stages(self, create_default_model: Model):
```
Comment (aronnoordam) [inline tests/test_stem.py:425]:
```suggestion
    def test_finalise_stages(self, create_default_model: Model):
```

File: tests/test_stem.py [hunk 2]
```diff
@@ -585,3 +618,101 @@ def test_set_output_name_new_stage_with_json_output(self, create_default_model:
 
         # check if the output name is set correctly
         assert new_stage.output_settings[0].output_name == "json_output_stage_3"
+
+    def test_finalise_stage_json_output(self):
```
Comment (aronnoordam) [inline tests/test_stem.py:622]: i dont think this test is needed, as you are testing the functionalities in test_model.py

Comment (morettid) [reply tests/test_stem.py:666]: The test is needed. The test in model checks the raising of errors but the the test in stem checks that the actual json file produced has keys in the right order.

File: tests/test_model.py [hunk 3]
```diff
@@ -3463,6 +3471,86 @@ def test_split_model_part_expected_raise(self, create_default_2d_soil_material:
         with pytest.raises(ValueError, match=f"Model part type and new parameters type must match."):
             model.split_model_part("process_2d", "split_group", [1], create_default_2d_soil_material)
 
+    def test_finalise(self, create_default_2d_soil_material: SoilMaterial):
```
Comment (aronnoordam) [inline tests/test_model.py:3474]: can you rename it to something like test_finalise_json_output_raises

Comment (morettid) [reply tests/test_model.py]: Done!

**Review by aronnoordam — APPROVED** [review]

**Review by aronnoordam — APPROVED** [review]

**Pull request merged**